### DISCUSSION TITLE
dropbox.com: nuisance

### DISCUSSION THREAD

**Discussion by DimitrisKalaboukas** [uBlockOrigin/uAssets#22280] [Q&A] [answered]
### Prerequisites

- [X] This is **NOT** a [_YouTube_](https://www.github.com/uBlockOrigin/uAssets/issues/20586), [_Facebook_](https://www.github.com/uBlockOrigin/uAssets/issues/3367) or [_Twitch_](https://www.github.com/uBlockOrigin/uAssets/issues/5184) report. _These sites MUST be reported by clicking their **respective links**_.
- [X] I read and understand the [policy about what is a valid filter issue](https://github.com/uBlockOrigin/uAssets/blob/master/README.md#uassets).
- [X] I verified that this issue is not a duplicate. (Use this [button](https://user-images.githubusercontent.com/585534/146582579-c32707a0-36a1-4cc5-ad50-83172c9f67a8.png) to find out.)
- [X] I did not remove any of the [default filter lists](https://github.com/uBlockOrigin/uAssets/assets/66517106/6e8d024a-5869-47cf-b142-b2357eb6c7d0), or I have verified that the issue was not caused by removing any of the default lists.
- [X] I did not enable additional filter lists, or I have verified that the issue still occurs without enabling additional filter lists.
- [X] I do not have custom filters/rules, or I have verified that the issue still occurs without custom filters/rules.
- [X] I am not using uBlock Origin (uBO) along with other content blocker extensions.
- [X] I have verified that the web browser's built-in blocker or [DNS blocking](https://en.wikipedia.org/wiki/DNS_blocking) (standalone or through a VPN) is not causing the issue.
- [ ] I did not answer truthfully to **ALL** the above checkboxes.

### URL address of the web page

`https://www.dropbox.com/scl/fi/w2s3zpk18ucdems51xey9/.mp3?rlkey=mydjva765wib05bbhmt7gf80x&dl=0`

### Category

nuisance

### Description

Intrusive prompts to install app (which isn’t required to view or download the file) and to sign up while downloading the file (which isn’t necessary either) on Firefox for Android. If these annoyances don’t meet your blocking criteria, please provide me with custom rules.

### Other extensions used

none

### Screenshot(s)

<details>
<summary>Screenshot(s)</summary>

![Screenshot_20240131_111207_Firefox](https://github.com/uBlockOrigin/uAssets/assets/101707608/29fc6c31-a72e-4e9e-b5db-6598ed11b75f)

![Screenshot_20240131_111707_Firefox](https://github.com/uBlockOrigin/uAssets/assets/101707608/dc020239-6a2e-4ded-a09c-ad9a01b5e12a)

</details>

### Configuration

<details>

```yaml
uBlock Origin: 1.55.0
Firefox: 115
filterset (summary):
 network: 420847
 cosmetic: 264978
 scriptlet: 49650
 html: 2266
listset (total-discarded, last-updated):
 added:
  https://adguardteam.github.io/AdGuardSDNSFilter/Filters/filter.txt: 61138-2317, 29m
  https://raw.githubusercontent.com/DandelionSprout/adfilt/master/AnnoyancesList: 94281-1940, 29m
  https://raw.githubusercontent.com/hoshsadiq/adblock-nocoin-list/master/nocoin.txt: 672-0, 29m
  https://www.dropbox.com/s/kqbpdbxeeieazhe/Custom%20filters.txt?dl=1: 320-0, now
  https://www.i-dont-care-about-cookies.eu/abp/: 25840-19, now
  adguard-generic: 81942-7728, now
  adguard-mobile: 9034-140, now
  adguard-spyware-url: 1435-104, now
  adguard-spyware: 90834-29312, now
  block-lan: 56-0, now
  curben-phishing: 87855-35, 30m
  adguard-social: 22082-851, now
  [14 lists not shown]: [too many]
 default:
  user-filters: 0-0, never
  ublock-filters: 37424-564, 30m Δ
  ublock-badware: 7808-10, 30m Δ
  ublock-privacy: 988-2, 30m Δ
  ublock-unbreak: 2246-0, 30m Δ
  ublock-quick-fixes: 139-0, 30m Δ
  easylist: 80394-34774, 30m Δ
  easyprivacy: 43604-3272, 30m Δ
  urlhaus-1: 8415-0, 30m
  plowe-0: 3786-1, now
  GRC-0: 1055-2, 29m
filterset (user): [empty]
trustedset:
 added: [array of 1 redacted]
userSettings:
 advancedUserEnabled: true
hiddenSettings:
 trustedListPrefixes: ublock- user-
supportStats:
 allReadyAfter: 60221 ms (selfie)
 maxAssetCacheWait: 6483 ms
 cacheBackend: indexedDB
popupPanel:
 blocked: 54
 network:
  dropbox.com: 50
  127.0.0.1: 3
  litix.io: 1
 extended:
  ##._mini-modal-inner-container_5qqo3_8
  ###consent-iframe
  ##.dig-Banner
  ##.top-notification-bar-container
  ##[class*="ShareButton" i]:not(button, body, html, article, [cla…
  ##+js(set-session-storage-item, has_seen_interstitial, true)
  ##+js(set-cookie, seen-sl-download-modal, true)
  ##+js(set-cookie, seen-sl-signup-modal, true)
```
</details>

**Comment by partingscientist** [comment]
Can you test the following?
```adblock
dropbox.com##+js(set-session-storage-item, has_seen_interstitial, true)
dropbox.com##+js(trusted-set-local-storage-item, dbx_previews_log_out_download_modal, 2147483647000)
```

Also, a maintainer might find this interesting.
> allReadyAfter: 60221 ms (selfie)

**Reply by DimitrisKalaboukas** [reply]
@partingscientist The first filter along with:

`dropbox.com#?#div[class^="_headerContainer"] > div[style]:has(> div[class^="_banner_"] > div[class^="_text-container"] > button > span:contains(app))`

are currently included in AdGuard Annoyances – Mobile App Banners list, but they don’t solve the issue on Android.

**Reply by krystian3w** [reply]
That filter cut small blue bar above "player".

If AdGuard reject Greek language then we need these filter with corrected word or all sentence/phrase:
```adb
! bar
dropbox.com#?#div[class^="_headerContainer"] > div[style]:has(> div[class^="_banner_"] > div[class^="_text-container"] > button > span:contains(Εμφάνιση στην εφαρμογή))
```

to hide blue bar.

**Comment by DimitrisKalaboukas** [comment]
@krystian3w ?

**Reply by krystian3w** [reply]
I do not use the website and no one has asked Polish Annoyances Filters to effectively hide/block it.

**Reply by DimitrisKalaboukas** [reply]
No account is required to reproduce the issue on Android.

**Reply by krystian3w** [reply]
I was more interested in the fact that maybe once 'every' 5 years I have to download something from this "cloud".

Since an account is not required, almost everyone can think of a solution when these with `set-session-storage-item` & `trusted-set-local-storage-item` there are not enough effects.

**Reply by krystian3w** [reply, marked as answer]
At my side on real smartphone with **partingscientist** scriptlets we can hide big last modal/pop-up after correct word/words used in header:

```adb
! May needed base at this time:
dropbox.com##+js(set-session-storage-item, has_seen_interstitial, true)
dropbox.com##+js(trusted-set-local-storage-item, dbx_previews_log_out_download_modal, 2147483647000)

! pop-up - use only one filter
dropbox.com##.ReactModalPortal:has([role="dialog"] h2:has-text(Το αρχείο σας κατεβαίνει))
dropbox.com##.ReactModalPortal .ReactModal__Overlay:has([role="dialog"] h2:has-text(Το αρχείο σας κατεβαίνει))

! useless empty div?
dropbox.com##div[data-react-modal-body-trap=""]:is([style*="opacity: 1"], [style*="opacity: ."], [style*="opacity: 0.]):empty:has(~ .ReactModalPortal [role="dialog"] h2:has-text(Το αρχείο σας κατεβαίνει))
```


due click on 'arrow' ⬇️ button at page saves files without close pop-up with girl on 'Garden chair' near 'rainforest'.

**Reply by DimitrisKalaboukas** [reply]
Which is the trusted list prefix to make custom filter lists trusted by uBO? The prefix “user-” applies only to “My filters”.

**Reply by krystian3w** [reply]
Long URLs separated by spaces e.g.:

` https://raw.githubusercontent.com/DimitrisKalaboukas/ https://gist.githubusercontent.com/DimitrisKalaboukas/`
